crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -295,6 +300,19 @@ impl TimelineState {
         result
     }
 
+    pub(super) fn reload_shields(&mut self) {
+        let Ok(is_room_encrypted) = self.meta.is_room_encrypted.read() else {
+            return;
+        };
+
+        if let Some(is_encrypted) = *is_room_encrypted {
```

Comment:
Don't we need to do this only if it's `Some()` as well as `true`? On the other hand, the only place where we call this is in a place where we know the value of this and if it'll be true, why don't we just unconditionally call `replace_all_events_encryption()`?